lib/galaxy/webapps/galaxy/api/authnz.py

```diff
@@ -0,0 +1,140 @@
+"""
+OpenID Connect (OIDC) Authentication and Authorization (authnz) API.
+"""
+
+from __future__ import absolute_import
+
+import logging
+
+from galaxy import web
+from galaxy.exceptions import MessageException
+from galaxy.web import (
+    _future_expose_api_anonymous_and_sessionless as expose_api_anonymous_and_sessionless
+)
+from galaxy.web import url_for
+from galaxy.web.base.controller import BaseAPIController
+
+log = logging.getLogger(__name__)
+
+
+class OIDC(BaseAPIController):
+
+    @web.expose
+    @web.require_login("list third-party identities")
+    def index(self, trans, **kwargs):
+        """
+        GET /api/authnz/
+            returns a list of third-party identities associated with the user.
+
+        :type  trans: galaxy.web.framework.webapp.GalaxyWebTransaction
+        :param trans: Galaxy web transaction.
+
+        :param kwargs: empty dict
+
+        :rtype: list of dicts
+        :return: a list of third-party identities associated with the user account.
+        """
+        rtv = []
+        for authnz in trans.user.social_auth:
+            rtv.append({'id': trans.app.security.encode_id(authnz.id), 'provider': authnz.provider})
+        return rtv
+
+    @expose_api_anonymous_and_sessionless
+    def login(self, trans, provider):
+        """
+        GET /api/authnz/{provider}/login
+            returns a URL to be used to initiate a user authentication on the given provider.
+
+        :type  trans: galaxy.web.framework.webapp.GalaxyWebTransaction
+        :param trans: Galaxy web transaction.
+
+        :type  provider: string
+        :param provider: The name of the provider which should be used to authenticate a user.
+
+        :rtype:  string
+        :return: A URL that should be to initiated a user authentication process.
+        """
+        if not trans.app.config.enable_oidc:
+            raise MessageException(
```

Comment:
Should probably throw a ConfigDoesNotAllowException here instead. https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/exceptions/__init__.py#L133